src/DataTable/utils/getVisibleColumns.jsx

```diff
@@ -35,12 +37,22 @@ export const selectColumn = {
   // Proptypes disabled as this prop is passed in separately
   /* eslint-disable react/prop-types */
   Cell: ({ row }) => {
+    const {
+      isSelectable, maxSelectedRows, onMaxSelectedRows, state: { selectedRowIds },
+    } = useContext(DataTableContext);
     const updatedProps = useConvertIndeterminateProp(row.getToggleRowSelectedProps());
+    const { index } = row;
+    const isRowSelected = index in selectedRowIds;
+    const selectedRowsLength = Object.keys(selectedRowIds).length;
+    const hasMaxSelectedRows = maxSelectedRows && maxSelectedRows === selectedRowsLength;
+    const disableCheck = isSelectable && hasMaxSelectedRows && !isRowSelected;
+    if (hasMaxSelectedRows && selectedRowIds[index]) { onMaxSelectedRows?.(); }
```

Comment:
Given we're working in a component passed to the `Cell` arg, we should be wary of React re-renders. If this component re-renders with the same values, would `onMaxSelectedRows` get called unnecessarily?

[curious] Should this be wrapped in a `useEffect` so that `onMaxSelectedRows` only gets called when `hasMaxSelectedRows` changes from `false` -> `true`, for example?